src/openfl/_internal/text/TextEngine.hx

```diff
@@ -899,35 +901,40 @@ class TextEngine {
 
 					case JUSTIFY:
 						if (lineWidths[lineIndex] < totalWidth) {
-							lineLength = 1;
-
+							spacesInLine = 1;
+							layoutGroupsInLine = 1;
 							for (j in (i + 1)...layoutGroups.length) {
 								if (layoutGroups[j].lineIndex == lineIndex) {
 									if (j == 0 || text.charCodeAt(layoutGroups[j].startIndex - 1) == " ".code) {
-										lineLength++;
+										spacesInLine++;
 									}
+									layoutGroupsInLine++;
 								} else {
 									break;
 								}
 							}
 
-							if (lineLength > 1) {
-								group = layoutGroups[i + lineLength - 1];
-
+							if (spacesInLine > 1) {
+								group = layoutGroups[i + layoutGroupsInLine - 1];
 								var endChar = text.charCodeAt(group.endIndex);
+								// Normal line ending if it's the end of a) text or b) paragrpah - No spreading out of the words.
 								if (group.endIndex < text.length && endChar != "\n".code && endChar != "\r".code) {
-									offsetX = (totalWidth - lineWidths[lineIndex]) / (lineLength - 1);
-
-									var j = 0;
+									offsetX = (totalWidth - lineWidths[lineIndex]) / (spacesInLine - 1);
+									var j = 1;
+									var k = 1;
 									do {
-										if (j > 1 && text.charCodeAt(layoutGroups[j].startIndex - 1) != " ".code) {
-											layoutGroups[i + j].offsetX += (offsetX * (j - 1));
+										if (text.charCodeAt(layoutGroups[i + j].startIndex - 1) != " ".code) {
```

Comment:
What is the case here where there are 2(or more) layoutGroups split with non-space character but we need to apply offset to the the second layout group. Could you point me to an example case ?

Reply:
Sample case would be a hyphen. It will create a new layout group, but this layout group should be positioned without a a larger, "justify" spacing.
See the attached images, I think it explains this better than words. The condition is commented out in the second image. 
![image](https://user-images.githubusercontent.com/85608978/121682364-4ed24e80-cabc-11eb-83df-c967a53ca45a.png)
![image](https://user-images.githubusercontent.com/85608978/121682475-6dd0e080-cabc-11eb-9998-53c56529b875.png)

Reply:
Totally off topic, but I like reading my mails from all of you. What a nice time we had together, at least that's what I feel 😅